server/server.ts

```diff
@@ -103,4 +103,29 @@ app.get("/api/generateDaily",(req, res) => {
     new dailyEntriesModel(daily).save();
   }
   res.send({message:"generated 100 datas!"});
-})
\ No newline at end of file
+})
+
+//Below is a post request for the users to register
+app.post("/api/register", (req, res) => {
+  //console.log('req here:!!', req.body);
+  let userData = req.body;
+  let userReg = new userEntriesModel({
+    firstName: userData.firstName,
+    lastName: userData.lastName,
+    age: +userData.age,
+    gender: userData.gender.value,
+    email: userData.email,
+    phoneNumber: +userData.phoneNumber,
+    height: +userData.height,
+    weight: +userData.weight,
+    caloriesGoal: +userData.targetCalories,
+    waterGoal: +userData.targetWater,
+    createdTime: new Date(),
+    updatedTime:{},
+  });
+  //userReg.save();
+  res.send({
+    code: 200,
+    essmsg: "user successfully post a request"
+  });
```

Comment:
See Async example. Need to use stand Http codes for standard error handling

Reply:
updated.